src/emmy/expression/compile.cljc

```diff
@@ -111,6 +112,8 @@
                              (- ~'x (~'Math/floor ~'x)))
                      :fn (fn [^double x]
                            (- x (Math/floor x)))}
+   'aset-double #?(:clj {:sym 'clojure.core/aset-double :f aset-double}
```

Comment:
CRAZILY I think `aset-double` is much slower than `aset`:

```clj
(let [arr ^doubles (double-array 4)
      n 10000000]
  (time (dotimes [_ n]
          (aset-double arr 0 1.0)))
  (time (dotimes [_ n]
          (aset arr 0 1.0))))
"Elapsed time: 571.995333 msecs"
"Elapsed time: 6.592667 msecs"
```

I think because it is coercing its argument each time? I would recommend that we stick with `aset` here...

Reply:
I got off on the wrong track with this because `aset` reflection-warns unless the array is type-hinted doubles and the value is type-hinted double. Once that is done there is no reflection and no temptation to use `aset-double`